cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,11 +40,40 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+
+	Phases Phases
+
+	// The directory where certificates are stored
+	CertificatesDir string
+}
+
+type PhaseMetadata struct {
+	Annotations map[string]string
+}
+
+type Phases struct {
+	Certificates CertificatesPhase
+}
+
+type CertificatesPhase struct {
+	Metadata PhaseMetadata
+
+	// In the future, we may provide more options for generating certs
+	// For instance, we may want to provide integrations with something like Vault in the future for storing the certs
+	SelfSign *SelfSignCertificates
+}
+
+type SelfSignCertificates struct {
```

Comment:
What does this phase create? The only self signed certificate is the ca.crt which doesn't need alt names, ServiceSubnet, DNSDomain, or AdvertiseAddresses